### DISCUSSION TITLE
Hardware acceleration doesn't work in Firefox when running in Sandboxie

### DISCUSSION THREAD

**Discussion by W1K23** [sandboxie-plus/Sandboxie#3101] [Q&A Feedback]
For a long time, I was angry with Firefox for playing YouTube videos with a much higher CPU usage than Edge, Chrome. I thought Youtube was optimized for chromium. Today I finally found the cause of the problem. 
I always run Firefox in Sandboxie. **Hardware acceleration doesn't work in SandBoxie.** Therefore the CPU usage is 3 times higher and my computer gets very hot while playing 4K videos. When Firefox is started in the normal way, there is no high CPU usage and the notebook doesn't heat up, the fan doesn't spin.
**I tried Vivaldi running in Sandboxie and it uses hw acceleration normally.** Why? What's the difference?
How to turn on Firefox hardware acceleration in Sandboxie? 

**Firefox by default**, about:support , **Graphics** - Compositing: **WebRender** 
[Test video 4K, CPU usage 4-8%, 55°C](https://cdn.rios.hu/dl/upc/2023-05/29723_ffsiman.jpg)

**Firefox in Sandboxie, Graphics** - Compositing: **WebRender (software)** 
[Test video 4K, CPU usage 34%, 91°C](https://cdn.rios.hu/dl/upc/2023-05/29723_ffsboxieban2.jpg)

**Still no solution?**
[VP9 Hardware Acceleration Broken for Firefox and Edge](https://github.com/sandboxie-plus/Sandboxie/issues/742)

[I asked about this on the Xanasoft forum on Apr 28, 2023, but I didn't get any meaningful answers!](https://forum.xanasoft.com/threads/hardware-acceleration-doesnt-work-in-firefox-and-other-browsers-when-running-in-sandboxie.570/)
I can't believe there isn't a setting, e.g. in config.ini...
I've been using Sandboxie for about 10 years, but it's been a while I (and my family) can't use it normally because it slows down our new laptops. 
Please help!

**Comment by hg421** [comment]
Hardware acceleration in Firefox has been broken since Mozilla implemented GPU process sandboxing in FF 108 (see https://github.com/sandboxie-plus/Sandboxie/issues/2444).

It is possible to workaround this by setting `security.sandbox.gpu.level=0` in `about:config` (this disables FF's own sandboxing for the GPU process so it's only recommended to do so when running under Sandboxie anyways). Then normal Hardware accelation should work again (Webrender, haven't tested VP9).

**Reply by isaak654** [reply]
The lead developer is currently unaware of a proper fix.
If you're looking for outside help, that's all I know: https://github.com/sandboxie-plus/Sandboxie/issues/3102#issuecomment-1636748491
